Cesium3DTilesSelection/src/Tileset.cpp

```diff
@@ -1275,18 +1515,35 @@ static bool isVisibleInFog(double distance, double fogDensity) noexcept {
 //   * The tile has not yet been added to a load queue.
 Tileset::TraversalDetails Tileset::_visitTileIfNeeded(
     const FrameState& frameState,
+    ImplicitTraversalInfo&& implicitInfo,
     uint32_t depth,
     bool ancestorMeetsSse,
     Tile& tile,
     ViewUpdateResult& result) {
-  tile.update(frameState.lastFrameNumber, frameState.currentFrameNumber);
-  this->_markTileVisited(tile);
 
-  const Tileset* pTileset = tile.getTileset();
-  if (!pTileset) {
-    return TraversalDetails();
+  // If the tile is ContentLoaded, complete processing it and push it to Done
+  // unless we are waiting on a subtree load. We have to wait for the subtree
+  // to load so that we don't create fake (upsampled) children before knowing
+  // whether or not the tile has actual implicit children by looking at the
+  // subtree.
```

Comment:
Not the end of the world, but it's a shame we have to avoid finishing/rendering a tile that is ready to go just to wait for a subtree that will only be relevant for refinement. I'm guessing if you could see any easy way to avoid this, though, you would have already implemented it. Is that right?

Reply:
It does seem hard to work around, but actually the issue gets even more complicated once we actually avoid making content requests for implicit tiles that are marked as content unavailable (referencing above conversations): 

So firstly, the only tiles that have to wait like this are going to each be the root tile of some unloaded or loading subtree. The fact that this subtree root tile exists at this point, implies that the parent subtree had marked this subtree as available; note also that the spec states the root tile of each available subtree must be available. That is why this Tile object has already been created, _however_, we actually don't know the _content_ availability of the tile. So until the subtree is loaded, we can't commit to treating the tile's content as being unavailable, and we can't (or shouldn't) blindly request the content from the template uri, and of course we can't create implicit children since we don't know their availabilities either before the subtree is loaded. 

In the current state though, since we are blindly requesting content on every TILE_AVAILABLE implicit tile anyways, this blocking only serves to avoid accidentally creating fake, upsampled children prematurely.

Reply:
Ok, I think I understand. And in a way that makes this simpler. We can't load this tile's content _at all_ until we load its subtree, because the subtree tells us whether the tile has content. So we do have to wait for the subtree before we can render this tile, but it's essential waiting required by the spec, not accidental waiting as a result of our implementation.

Reply:
This logic ended up going away, because tiles should not even end up in the ContentLoaded state if their content availability is not known yet (the subtree hasn't loaded yet). `addTileToLoadQueue` will now refuse to move a tile out of `Unloaded` if the availability is still loading.